k3k-kubelet/provider/provider.go

```diff
@@ -608,6 +612,10 @@ func (p *Provider) updatePod(ctx context.Context, pod *v1.Pod) error {
 	currentHostPod.Spec.ActiveDeadlineSeconds = pod.Spec.ActiveDeadlineSeconds
 	currentHostPod.Spec.Tolerations = pod.Spec.Tolerations
 
+	// in the virtual cluster we can update also the labels and annotations
+	currentHostPod.Annotations = pod.Annotations
+	currentHostPod.Labels = pod.Labels
+
```

Comment:
It's probably safer to keep the previous annotations/labels, merging the two maps:

```go
maps.Copy(currentHostPod.Annotations, pod.Annotations)
maps.Copy(currentHostPod.Labels, pod.Labels)
```